src/CommandHandler/CancelPaymentHandler.php

```diff
@@ -39,6 +39,10 @@ public function __invoke(CancelPayment $command): void
             return;
         }
 
+        if(array_key_exists('expires_at', $payment->getDetails()) && $payment->getDetails()['expires_at'] < time()) {
```

Comment:
There is a service to get the current time in Symfony, can you use it instead of using `time()` function ? https://symfony.com/doc/current/components/clock.html#nativeclock

Using it allows for more control over the current date.

Reply:
```suggestion
        $details = $payment->getDetails();
        if(!isset($details['expires_at'])) {
            return;
        }
        
        if($details['expires_at'] < $this->clock->now()->getTimestamp()) {
```